extensions/ql-vscode/src/variant-analysis/code-scanning-pack.ts

```diff
@@ -25,20 +28,30 @@ export async function resolveCodeScanningQueryPack(
 
   // Resolve queries
   void logger.log(`Resolving queries for pack: ${packName}`);
-  const suitePath = join(
-    packDir,
-    "codeql-suites",
-    `${language}-code-scanning.qls`,
-  );
+
+  const suiteFile = await file({
+    postfix: ".qls",
+  });
+  const suitePath = suiteFile.path;
+  const suiteYaml: SuiteInstruction[] = [
+    {
+      import: `codeql-suites/${language}-code-scanning.qls`,
+      from: `${downloadedPack.name}@${downloadedPack.version}`,
+    },
+    // Exclude any non-problem queries
+    {
+      exclude: {
+        kind: ["diagnostic", "metric"],
+      },
+    },
+  ];
+  await outputFile(suitePath, dump(suiteYaml), "utf8");
+
   const resolvedQueries = await cliServer.resolveQueries(suitePath);
 
-  const problemQueries = await filterToOnlyProblemQueries(
-    logger,
-    cliServer,
-    resolvedQueries,
-  );
+  await suiteFile.cleanup();
```

Comment:
To avoid leaving this file around in the case of an error, could this be in a `finally` block?